test/serverExplorer.test.ts

```diff
@@ -0,0 +1,193 @@
+    };
+
+    const ProcessOutput: Protocol.ServerProcessOutput = {
+        processId: 'process id',
+        server: serverHandle,
+        streamType: 0,
+        text: 'the type'
+    };
+
+    const findServerBeans = {
+        length: 1,
+        fullVersion: 'version',
+        location: 'path',
+        name: 'EAP',
+        serverAdapterTypeId: 'org.jboss',
+        specificType: 'EAP',
+        typeCategory: 'EAP',
+        version: '7.1'
+    };
+
+    const status = {
+        code: 0,
+        message: 'ok',
+        pluginId: 'unknown',
+        severity: 0
+    };
+
+    test('InsertServer call should add server to tree data model', () => {
+        const refreshStub = sandbox.stub(serverExplorer, 'refresh').returns(null);
+        const getChildren = serverExplorer.getChildren(null);
+        serverExplorer.insertServer(serverHandle);
+        expect(refreshStub).calledOnce;
+        expect(getChildren).deep.equals(undefined);
+    });
+
+    test('removeServer call should remove server from tree data model', () => {
+        const disposeStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            clear: () => {},
+            dispose: () => {}
+        });
+        const getChildren = serverExplorer.getChildren(null);
+        sandbox.stub(serverExplorer, 'refresh');
+        serverExplorer.removeServer(serverHandle);
+        expect(disposeStub).calledOnce;
+        expect(getChildren).deep.equals(undefined);
+    });
+
+    test('serverExplorer.updateServer call should update server state to received in state change event', () => {
+        const serviceStub = sandbox.stub(serverExplorer.servers, 'get').returns({
+            stateChange
+        });
+        const clearStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            clear: () => {
+                return true;
+            }
+        });
+        sandbox.stub(serverExplorer, 'refresh');
+        const updateserver = serverExplorer.updateServer(stateChange);
+        expect(serviceStub).calledOnce;
+        expect(clearStub).calledOnce;
+        expect(clearStub).calledOnceWith(updateserver);
+    });
+
+    test('serverExplorer.showOutput call should show servers output channel', () => {
+        const showStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            show: () => {}
+        });
+        serverExplorer.showOutput(serverHandle);
+        expect(showStub).calledOnce;
+
+    });
+
+    test('serverExplorer.getTreeItem call should return TreeItem instance with corresponding label', () => {
+        const serverHandleStub = sandbox.stub(serverExplorer.serverStatus, 'get').resolves({
+            serverHandle
+        });
+        serverExplorer.getTreeItem(serverHandle);
+        expect(serverHandleStub).calledOnce;
+    });
+
+    test('should able to getChildren of the server', () => {
+        const clearStub = sandbox.stub(serverExplorer.servers, 'values').returns([]);
```

Comment:
Sorry, but this is nonsense :) You stub the `values` method of the servers HasMap to return empty array and save the stub in `clearStub' variable. Then  you call `getChildren` and save empty array to `getChildren` variable. Finally you transform the 'clearStub' to array with Array.from() and get the empty array. Then deep comparison of two empty arrays gives you passing test :)

Lets just remove this test.